packages/@lwc/ssr-compiler/src/__tests__/api-decorator.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { describe, test, expect } from 'vitest';
+import { compileComponentForSSR } from '../index';
+
+const compile =
+    (src: string, filename = 'test.js') =>
+    () => {
+        return compileComponentForSSR(src, filename, {});
+    };
+
+describe('thows error', () => {
+    test('combined with @track', () => {
+        const src = /* js */ `
+            import { api, track, LightningElement } from "lwc";
+            export default class Test extends LightningElement {
+                @track
+                @api
+                apiWithTrack = "foo";
+            }
+        `;
+        expect(compile(src)).toThrow(`LWC1093: @api method or property cannot be used with @track`);
+    });
```

Comment:
I manually replicated here all `babel-plugin-component` fixture tests that are related to api decorator errors. (see https://github.com/salesforce/lwc/issues/5032#issuecomment-2555743591)